src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,598 @@ const ActorInit En_Box_InitVars = {
     (ActorFunc)EnBox_Update,
     (ActorFunc)EnBox_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C8DC0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C8DC8.s")
+static AnimationHeader* D_809CA800[4] = { &D_0600024C, &D_06000128, &D_0600043C, &D_0600043C };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 0, ICHAIN_STOP),
+};
+
+static s32 unused;
+
+void EnBox_SetupAction(EnBox* this, EnBoxActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnBox_ClipToGround(EnBox* this, GlobalContext* globalCtx) {
+    f32 newY;
+    CollisionPoly* a1;
+    void* a2;
+    Vec3f pos;
+
+    pos = this->dyna.actor.posRot.pos;
+    pos.y += 1.0f;
+    newY = func_8003C9A4(&globalCtx->colCtx, &a1, &a2, &this->dyna.actor, &pos);
+    if (newY != -32000.0f) {
+        this->dyna.actor.posRot.pos.y = newY;
+    }
+}
+
+void EnBox_Init(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    EnBox* this = THIS;
+    AnimationHeader* animHeader;
+    s32 dynaUnk;
+    f32 animFrameStart;
+    f32 animFrameCount;
+    s32 linkAge;
+
+    animFrameStart = 0.0f;
+    linkAge = gSaveContext.linkAge;
+    animHeader = D_809CA800[linkAge];
```

Comment:
FYI this is probably just the `(0, foo)` pattern and there is no actual temp
```suggestion
    animHeader = D_809CA800[(0, gSaveContext.linkAge)];
```